common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -325,9 +325,19 @@ def test_get_user_group_id_for_partition(self):
 
         # switch to the second group and verify that it is returned for the user
         self.user_partition.scheme.current_group = groups[1]    # pylint: disable=no-member
-        group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #self.assertEqual(group2_id, groups[1].id)    # pylint: disable=no-member
+
+        partition_service2 = StaticPartitionService(
+            [self.user_partition],
+            user=Mock(username='ma', email='ma@edx.org', is_staff=False, is_active=True),
+            course_id=self.course.id,
+            track_function=Mock()
+        )
+        group2_id = partition_service2.get_user_group_id_for_partition(user_partition_id)
```

Comment:
But that's ok in terms of system behavior, right? Does the code rely on the fact that you can change a user's partitioning scheme and read it back in the same request?